src/main/java/Controller/newPlaylistController.java

```diff
@@ -0,0 +1,48 @@
+package Controller;
+import Entities.RegularUser;
+import Entities.Song;
+import InputBoundary.newPlaylistInputBoundary;
+import InputData.newPlaylistInputData;
+import UseCase.CreatePlaylistInteractor;
+import Presenter.playlistPresenter;
+
+/**
+ * Controller-layer implementation for create playlist use case
+ */
+public class newPlaylistController {
+    private final newPlaylistInputBoundary newPlaylistInputBoundary;
+
+//    private final newPlaylistInputData newPlaylistInputData;
+//
+//    private CreatePlaylistInteractor createPlaylist;
+
+    public newPlaylistController(CreatePlaylistInteractor createPlaylist){
```

Comment:
This isn't technically wrong, but for the sake of consistency I believe we were making the controller construct the interactor.

Reply:
You'd probably also want to take in the Presenter as an OutputBoundary (as you had commented out originally) to give to the Interactor

Reply:
Ah, I might have missed that then when re-working my code

Reply:
That would require creating a new playlist interactor object in the constructor and setting this.inputboundary to said object correct ?

Reply:
Also Gries' code does something similar but instead passes an input boundary object in the constructor instead of a createplaylist interactor